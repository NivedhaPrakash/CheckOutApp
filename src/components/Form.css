.demoForm {
	width: 500px;
	margin: auto;
	margin-top:5em;
}
.carou{
	height: 50em;
	width: 50em;
}
.bg{
	width: 100%;
	height: 88vh;
	background-position: center;
	background-repeat:no-repeat;
}