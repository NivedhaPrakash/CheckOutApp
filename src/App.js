import React, { Component } from 'react';
import { Router, Route, Switch } from 'react-router-dom';

import history from './history';
import './App.css';
import Form from './components/Form';
import Front from './components/Front';
import Carou from './components/Carou';
import Image from './components/Image';
import Errors from './components/Errors';
import Shopping from './components/Shopping';
import Login from './components/Login';

class App extends Component {
  handleLogin=(e)=>{
    e.preventDefault();
    history.push("/login");
  }
  handleHome=(e)=>{
    e.preventDefault();
    history.push("/");
  }
  handleSign=(e)=>{
    e.preventDefault();
    history.push("/form");
  }
  render() {
    return (

      <div className="App">
        <header className="App-header">
          <h1 className="App-title">Check Out App</h1>
          <button type="submit" className="btn btn-primary but"
           onClick={(e)=>this.handleHome(e)}>Home</button>
          <button type="submit" className="btn btn-primary but"
           onClick={(e)=>this.handleLogin(e)}>Login</button>
            <button type="submit" className="btn btn-primary but"
            onClick={(e)=>this.handleSign(e)}>Sign Up</button>
        </header>
        <Router history={history}>
          <Switch>
            <Route path="/" component={Front} exact />
            <Route path="/form" component={Form} exact />
            <Route path="/login" component={Login} exact />
            <Route path="/image" component={Image} exact />
            <Route path="/carousel" component={Carou} exact />
            <Route path="/shopping" component={Shopping} exact />
            <Route component={Errors} />
          </Switch>
        </Router>
      </div>

    );
  }
}

export default App;
