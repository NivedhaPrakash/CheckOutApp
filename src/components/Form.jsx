import React, { Component } from 'react';

import '../../src/App.css';
import history from '../history';
import { FormErrors } from './FormErrors.jsx';
import './Form.css';
import '../../src/App';

class Form extends Component {
  constructor (props) {
    super(props);
    this.state = {
      id:1,
      username: '',
      password: '',
      confrmpasswrd:'',
      formErrors: {username: '', password: '',confrmpasswrd:''},
      usernameValid: false,
      passwordValid: false,
      confrmValid:false,
      formValid: false
    }
  }

  handleUserInput = (e) => {
    const name = e.target.name;
    const value = e.target.value;
    this.setState({[name]: value},
                  () => { this.validateField(name, value) });
  }

  validateField(fieldName, value) {
    let fieldValidationErrors = this.state.formErrors;
    let usernameValid = this.state.usernameValid;
    let passwordValid = this.state.passwordValid;
    let confrmValid = this.state.confrmValid;

    switch(fieldName) {
      case 'username':
        usernameValid = value.match('^[a-z]{5,}$');
        fieldValidationErrors.username = usernameValid ? '' : ' is invalid';
        break;
      case 'password':
        passwordValid = value.length>6;
        fieldValidationErrors.password = passwordValid ? '': ' is too short';
        break;
      case 'confrmpasswrd':
        confrmValid = value.length>6;
        fieldValidationErrors.confrmpasswrd = confrmValid ? '': 'Not Matching With Passwrod';
        break;
      default:
        break;
    }
    this.setState({formErrors: fieldValidationErrors,
                    usernameValid: usernameValid,
                    passwordValid: passwordValid,
                    confrmValid: confrmValid,
                  }, this.validateForm);
  }

  validateForm() {
    this.setState({formValid: this.state.usernameValid && this.state.passwordValid && this.state.confrmValid});
  }

  errorClass(error) {
    return(error.length === 0 ? '' : 'has-error');
  }

  handleSubmit=(e)=>{
    this.setState({id:++this.state.id});
    console.log(this.state.id);
    e.preventDefault();
    fetch('http://localhost:4000/users',{
      method:'POST',
      headers:{
        Accept:'application/json',
        'Content-Type':'application/json',
      },
      body:JSON.stringify({
        id:this.state.id,
        username:this.state.username,
        password:this.state.password,
      }),
    });
    history.push("/image");
    console.log(this.state.id+"-->"+this.state.username+"-->"+this.state.password);
  }
 render () {
   return (
  <form className="demoForm jumbotron container">
    <h2>Sign up</h2>
    <br/>
    <div>
      <FormErrors formErrors={this.state.formErrors} />
    </div>
    <div className={`form-group ${this.errorClass(this.state.formErrors.username)}`}>
      <input type="text" required className="form-control" name="username"
        placeholder="UserName"
        value={this.state.username}
        onChange={this.handleUserInput}  />
    </div>
    <div className={`form-group ${this.errorClass(this.state.formErrors.password)}`}>
      <input type="password" className="form-control" name="password"
        placeholder="Password"
        value={this.state.password}
        onChange={this.handleUserInput}  />
    </div>
    <div className={`form-group ${this.errorClass(this.state.formErrors.confrmpasswrd)}`}>
      <input type="password" className="form-control" name="confrmpasswrd"
        placeholder="Confirm Password"
        value={this.state.confrmpasswrd}
        onChange={this.handleUserInput}  />
    </div>
    <button type="submit" className="btn btn-primary button-click" disabled={!this.state.formValid} onClick={(e)=>this.handleSubmit(e)}>Sign up</button>
  </form>
   )
 }
}
export default Form;