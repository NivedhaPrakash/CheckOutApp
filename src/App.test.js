import React from 'react';
import ReactDOM from 'react-dom';
import { shallow,mount } from 'enzyme';
import App from './App';
import Form from './components/Form';

const localStorageMock = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  clear: jest.fn()
};
global.localStorage = localStorageMock;

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Form', () => {
  it('should be defined', () => {
    expect(Form).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(
      <Form name='button test' />
    );
    expect(tree).toMatchSnapshot();
  });
 });

 
