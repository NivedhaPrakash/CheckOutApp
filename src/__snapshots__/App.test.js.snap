// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form
    name="button test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Sign up
</h2>,
        <br />,
        <div>
          <Unknown
                    formErrors={
                              Object {
                                        "confrmpasswrd": "",
                                        "password": "",
                                        "username": "",
                                      }
                    }
          />
</div>,
        <div
          className="form-group "
>
          <input
                    className="form-control"
                    name="username"
                    onChange={[Function]}
                    placeholder="UserName"
                    required={true}
                    type="text"
                    value=""
          />
</div>,
        <div
          className="form-group "
>
          <input
                    className="form-control"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
          />
</div>,
        <div
          className="form-group "
>
          <input
                    className="form-control"
                    name="confrmpasswrd"
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    type="password"
                    value=""
          />
</div>,
        <button
          className="btn btn-primary button-click"
          disabled={true}
          onClick={[Function]}
          type="submit"
>
          Sign up
</button>,
      ],
      "className": "demoForm jumbotron container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign up",
        },
        "ref": null,
        "rendered": "Sign up",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Unknown
            formErrors={
                        Object {
                                    "confrmpasswrd": "",
                                    "password": "",
                                    "username": "",
                                  }
            }
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "formErrors": Object {
              "confrmpasswrd": "",
              "password": "",
              "username": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="form-control"
            name="username"
            onChange={[Function]}
            placeholder="UserName"
            required={true}
            type="text"
            value=""
/>,
          "className": "form-group ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "name": "username",
            "onChange": [Function],
            "placeholder": "UserName",
            "required": true,
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="form-control"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
/>,
          "className": "form-group ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="form-control"
            name="confrmpasswrd"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
            value=""
/>,
          "className": "form-group ",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "name": "confrmpasswrd",
            "onChange": [Function],
            "placeholder": "Confirm Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign up",
          "className": "btn btn-primary button-click",
          "disabled": true,
          "onClick": [Function],
          "type": "submit",
        },
        "ref": null,
        "rendered": "Sign up",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Sign up
</h2>,
          <br />,
          <div>
            <Unknown
                        formErrors={
                                    Object {
                                                "confrmpasswrd": "",
                                                "password": "",
                                                "username": "",
                                              }
                        }
            />
</div>,
          <div
            className="form-group "
>
            <input
                        className="form-control"
                        name="username"
                        onChange={[Function]}
                        placeholder="UserName"
                        required={true}
                        type="text"
                        value=""
            />
</div>,
          <div
            className="form-group "
>
            <input
                        className="form-control"
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
</div>,
          <div
            className="form-group "
>
            <input
                        className="form-control"
                        name="confrmpasswrd"
                        onChange={[Function]}
                        placeholder="Confirm Password"
                        type="password"
                        value=""
            />
</div>,
          <button
            className="btn btn-primary button-click"
            disabled={true}
            onClick={[Function]}
            type="submit"
>
            Sign up
</button>,
        ],
        "className": "demoForm jumbotron container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign up",
          },
          "ref": null,
          "rendered": "Sign up",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Unknown
              formErrors={
                            Object {
                                          "confrmpasswrd": "",
                                          "password": "",
                                          "username": "",
                                        }
              }
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "formErrors": Object {
                "confrmpasswrd": "",
                "password": "",
                "username": "",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control"
              name="username"
              onChange={[Function]}
              placeholder="UserName"
              required={true}
              type="text"
              value=""
/>,
            "className": "form-group ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "name": "username",
              "onChange": [Function],
              "placeholder": "UserName",
              "required": true,
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
/>,
            "className": "form-group ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control"
              name="confrmpasswrd"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value=""
/>,
            "className": "form-group ",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "name": "confrmpasswrd",
              "onChange": [Function],
              "placeholder": "Confirm Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign up",
            "className": "btn btn-primary button-click",
            "disabled": true,
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "Sign up",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
